File: cpp/bc1145.cpp
#include <iostream>

using namespace std;

int main()
{
	int count = 0, fVal, sVal;
	
	cin >> fVal >> sVal;
	
	cout << 1;
	
	for(int i = 2; i != sVal; i++)
	{
		cout << " " << i;
		count++;
		
		if(count == fVal-1)
		{
			cout << endl;
			cout << i+1;
			count = 0;
			i++;
		}
	}
	
	cout << " " << sVal << endl;
}

File: python/bc1145.py
cVal = input().split(" ")
count = 0

fVal = int(cVal[0])
sVal = int(cVal[1])

print(1, end="")

for i in range(2, sVal):
    if count == fVal-1:
        print(f"\n{i}", end="")
        count = 0
    else:
        print(f" {i}", end="")
        count += 1

print(f" {sVal}")